### PR TITLE
CERES-783: use cassandra for metric metadata dedup check

### PR DESCRIPTION
**Pull request by iWebi**
# Resolves

https://rackspace.atlassian.net/browse/CERES-783

# What

Replace dedup check implementation to use Cassandra instead of Redis. Redis keys were not being TTLd.
We could add TTL to Redis#setIfAbsent calls but that leads to many keys being TTLd. Also the same data already exists in Cassandra. Opting to use  Cassandra#exists check instead for the need

# How

Replace Redis#setIfAbsent calls with Cassandra#exists check

## How to test

Unit tests+Dev cluster Grafana stats

# Why

Alternately we could continue to use Redis#setIfAbsent + TTL parameter.

# TODO

None

### CODE REVIEW

File: src/main/java/com/rackspace/ceres/app/services/MetadataService.java [hunk 1]
```diff
@@ -123,31 +113,32 @@ public MetadataService(ReactiveCqlTemplate cqlTemplate,
...
-                .flatMap(inserted -> !inserted ? Mono.just(true) :
-                    // not cached, so store the metadata to be sure
-                    storeMetadataInCassandra(tenant, seriesSetHash, metricName,
-                        tags
-                    )
-                        .map(it -> true))
-                .toFuture()
+            new SeriesSetCacheKey(tenant, seriesSetHash),
+            (key, executor) ->
+                    cassandraTemplate.exists(
```
Comment (rohitsngh27) [inline src/main/java/com/rackspace/ceres/app/services/MetadataService.java:118]: What if we remove this check from here and simply put it in database if metadata is note present in in memory cache?

**Review by rohitsngh27 — APPROVED** [review]
LGTM.

File: src/main/java/com/rackspace/ceres/app/services/MetadataService.java (hunk 1 above)
Comment (matsboman) [reply src/main/java/com/rackspace/ceres/app/services/MetadataService.java:118]: Either way is fine I think. This is a very good improvement.

**Review by matsboman — APPROVED** [review]
LGTM

**Pull request merged**